cyclonedx_py/parser/requirements.py

```diff
@@ -17,48 +17,66 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) OWASP Foundation. All Rights Reserved.
 
+import os
+import os.path
+from tempfile import NamedTemporaryFile, _TemporaryFileWrapper  # Weak error
+from typing import Any, Optional
+
 from cyclonedx.model.component import Component
+from cyclonedx.model import HashType
 from cyclonedx.parser import BaseParser, ParserWarning
 # See https://github.com/package-url/packageurl-python/issues/65
 from packageurl import PackageURL  # type: ignore
-from pkg_resources import parse_requirements as parse_requirements
+from pip_requirements_parser import RequirementsFile  # type: ignore
 
 
 class RequirementsParser(BaseParser):
 
     def __init__(self, requirements_content: str) -> None:
         super().__init__()
+        parsed_rf: Optional[RequirementsFile] = None
+        requirements_file: Optional[_TemporaryFileWrapper[Any]] = None
 
-        requirements = parse_requirements(requirements_content)
-        for requirement in requirements:
-            """
-            @todo
-            Note that the below line will get the first (lowest) version specified in the Requirement and
-            ignore the operator (it might not be ==). This is passed to the Component.
+        if os.path.exists(requirements_content):
+            parsed_rf = RequirementsFile.from_file(
+                requirements_content, include_nested=True)
+        else:
+            requirements_file = NamedTemporaryFile(mode='w+', delete=False)
+            requirements_file.write(requirements_content)
+            requirements_file.close()
 
-            For example if a requirement was listed as: "PickyThing>1.6,<=1.9,!=1.8.6", we'll be interpreting this
-            as if it were written "PickyThing==1.6"
-            """
-            try:
-                (op, version) = requirement.specs[0]
-                self._components.append(Component(
-                    name=requirement.project_name, version=version, purl=PackageURL(
-                        type='pypi', name=requirement.project_name, version=version
-                    )
-                ))
-            except IndexError:
+            parsed_rf = RequirementsFile.from_file(
+                requirements_file.name, include_nested=False)
+
+        for requirement in parsed_rf.requirements:
+            name = requirement.link.url if requirement.is_local_path else requirement.name
+            version = (
+                requirement.get_pinned_version
+                if requirement.get_pinned_version
+                else requirement.dumps_specifier())
+            hashes = [HashType.from_composite_str(hash) for hash in requirement.hash_options]
```

Comment:
more pythonic? 

```patch
-              hashes = [HashType.from_composite_str(hash) for hash in requirement.hash_options]
+              hashes = map(HashType.from_composite_str, requirement.hash_options)
```

not even needed to create a list from that `map()`, since `Component(hashes=...) accepts all sorts of iterables(generators) 


hey, @madpah , this is why you did that implementation last month, so we can finally pull of such cool tricks :)